app/forms/hub/take_action_form.rb

```diff
@@ -40,14 +48,86 @@ def contact_method_help_text
+      raise TakeActionFormError.new("Can't initiate status change to current status.") if status == tax_return.status
+
+      if tax_return.update!(status: status)
+        @action_list << I18n.t("hub.clients.update_take_action.flash_message.status")
+        SystemNote.create_status_change_note(@current_user, tax_return)
+        send_message if message_body.present?
+        create_note if internal_note_body.present?
+      end
+    end
+
     def self.permitted_params
-      [:tax_returns, :locale, :message_body, :contact_method, :internal_note_body, { tax_returns: {} }]
+      [:tax_return_id, :status, :locale, :message_body, :contact_method, :internal_note_body]
     end
 
     private
 
+    def send_message
+      case contact_method
+      when "email"
+        send_email(message_body, subject_locale: locale)
+        @action_list << I18n.t("hub.clients.update_take_action.flash_message.email")
+      when "text_message"
+        send_text_message(message_body)
+        @action_list << I18n.t("hub.clients.update_take_action.flash_message.text_message")
+      end
+    end
+
+    def create_note
+      Note.create!(
+          body: internal_note_body,
+          client: @client,
+          user: @current_user
+      )
+      @action_list << I18n.t("hub.clients.update_take_action.flash_message.internal_note")
+    end
+
+    def current_user
+      @current_user
+    end
+
     def language_label(key)
       I18n.t("general.language_options.#{key}")
     end
+
+    def set_default_message_body
+      @message_body = case status
+                      when "intake_more_info", "prep_more_info", "review_more_info"
+                        document_list = @client.intake.relevant_document_types.map do |doc_type|
+                          "  - " + doc_type.translated_label(@client.intake.locale)
+                        end.join("\n")
+                        I18n.t(
+                            "hub.status_macros.needs_more_information",
+                            required_documents: document_list,
+                            document_upload_link: @client.intake.requested_docs_token_link,
+                            locale: @client.intake.locale
+                        )
+                      when "prep_ready_for_review"
+                        I18n.t("hub.status_macros.ready_for_qr", locale: @client.intake.locale)
+                      when "filed_accepted"
+                        I18n.t("hub.status_macros.accepted", locale: @client.intake.locale)
+                      else
+                        ""
+                      end
+    end
+
+    def set_default_contact_method
+      default = "email"
+      prefers_sms_only = @client.intake.sms_notification_opt_in_yes? && @client.intake.email_notification_opt_in_no?
+      @contact_method = prefers_sms_only ? "text_message" : default
+    end
+
+    def set_default_locale
+      @locale = @client.intake.locale
+    end
   end
 end
+
+class TakeActionFormError < StandardError
```

Comment:
See notes later-on, in the test suite, for how we can avoid needing this exception if the form uses `.save`.